### ISSUE TITLE
Support collecting C/C++ configuration in Unreal Engine projects

### ISSUE THREAD

**Issue by rita-gorokhod** [SonarSource/sonarlint-visualstudio#2127]
#### Description

See community request: https://community.sonarsource.com/t/sonarlint-with-c-using-unreal-engine-4/37303

Unreal Engine projects are using Makefile built with NMake.

Prerequisite: supporting makefiles #1546

**Comment by Bdoom** [comment]
Hope to see this in the next version.

**Comment by Johannes-Knippel** [comment]
Hoping to see this coming soon!

**Comment by Bdoom** [comment]
> 
> 
> Hoping to see this coming soon!

It's been confirmed for the next release I believe.

**Comment by Johannes-Knippel** [comment]
> > Hoping to see this coming soon!
> 
> It's been confirmed for the next release I believe.

Lovely. When will the next release be? Somehow I can't find anything on that.

**Comment by marco-comi-sonarsource** [comment]
Hello,
at SonarSource we are definitely tracking this request, and we will keep it into account for our next roadmap decisions; on the other hand, be advised that we will have to weight this request among many other ideas for improvement and new functionalities, so we do not expect to see this delivered in a short time frame. Just for the sake of clarity, this is not in scope for the next SonarLint for Visual Studio release.

**Comment by Bdoom** [comment]
> 
> 
> Hello,
> at SonarSource we are definitely tracking this request, and we will keep it into account for our next roadmap decisions; on the other hand, be advised that we will have to weight this request among many other ideas for improvement and new functionalities, so we do not expect to see this delivered in a short time frame. Just for the sake of clarity, this is not in scope for the next SonarLint for Visual Studio release.

I must have been misinformed then. I do think this will be a very massive addition and should be ranked highly against other additional feature adds. Why? Because Unreal Engine has a large set of developers that you are missing out on. There are entire chat rooms which could be encouraged to use this tool with game developers and companies. There are over 5,000 people on chat server alone that I am apart of where about 80% of the people are engineers. I think it is in your best interests to add this as you would be gaining access to an entire market of new developers who may choose to purchase your other products such as SonarQube and SonarCloud. Your entry level entrance to Sonar's products is SonarLint for VS. Gaining more traction in the game dev space will make you stand out massively. There are no other high quality linting tools for unreal engine. You would essentially corner the market and own the market. 

However, if you still believe it is of not a high priority, then that makes sense as well. There must be other features which would help you to obtain higher levels of additional potential customers for the company.

**Comment by Johannes-Knippel** [comment]
The company I work for is using SonarQube and we are around 7000 Developers using it in our daily development for a big variety of projects and products. The amount of Unreal developers is yet not that big (around 150) but we grow quite fast because the Engine helps us to virtualize eingineering, planning and design processes. 
This feature should definetly be part of your product. I used it for other code projects and really liked the way it helps you to improve code quality.
Hope you can reprioritize this feature.

**Comment by Bdoom** [comment]
> 
> 
> The company I work for is using SonarQube and we are around 7000 Developers using it in our daily development for a big variety of projects and products. The amount of Unreal developers is yet not that big (around 150) but we grow quite fast because the Engine helps us to virtualize eingineering, planning and design processes.
> This feature should definetly be part of your product. I used it for other code projects and really liked the way it helps you to improve code quality.
> Hope you can reprioritize this feature.

Yeah my company uses SonarQube and SonarLint for our project. Not nearly as big, we only have about 80~ or so people. But it has no unreal support. We use SonarLint for our C# projects though and it works great.

**Comment by guitarjorge24** [comment]
At my company we have several employees trying to use this as the main and official linting tool for our project. However, as it doesn't work on UE4 we may have to look for other solutions and have to cancel our subscription with SonarQube as we need a solution that integrates into all of our projects, not just our C# ones.

**Comment by martinsweitzer** [comment]
Still not supporting Unreal Engine?

**Comment by Staunch8801** [comment]
Can confirm that large engineering organizations are actively looking for static analysis tools for unreal.  This would be really good.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [SLVS-2259](https://sonarsource.atlassian.net/browse/SLVS-2259)

[SLVS-2259]: https://sonarsource.atlassian.net/browse/SLVS-2259?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
